app/src/main/java/org/fossasia/openevent/app/contract/model/IEventRepository.java

```diff
@@ -1,18 +1,22 @@
 package org.fossasia.openevent.app.contract.model;
 
+import org.fossasia.openevent.app.data.models.Attendee;
 import org.fossasia.openevent.app.data.models.Event;
 import org.fossasia.openevent.app.data.models.User;
 
 import java.util.List;
 
 import io.reactivex.Observable;
 
-public interface IEventModel {
+public interface IEventRepository {
```

Comment:
Actually Repository is an interface, it is a model which abstracts the pattern of loading data from multiple sources, namely disk, cache or network

However, your second point is very correct. It should be just EventDataRepository

Why we want to have same name is that DataSource != DataRepository

DataSource is like NetworkDataSource DiskDataSource, so when we create a persistent model, it will class with the repository interface